WordPress/gutenberg
# Query Loop Block: Use BlockPatternsList component for Query block patterns modal

Fixes https://github.com/WordPress/gutenberg/issues/33202

## What?

Replace the `BlockPatternSetup` component with `BlockPatternsList` component to match the experience of selecting patterns in different locations across the editor. 

## Why?

Lots of feedback on the linked issue above. In a nutshell the current carousel style pattern selection feels broken and confusing to users because it is different from anything else.

By moving to `BlockPatternsList` we start the process of consolidating around that component and moving closer to a [consistent experience for pattern selection](https://github.com/WordPress/gutenberg/issues/39308#issuecomment-1133086932).

## How?

This PR separates out the pattern selection modal functionality and replaces it to use the `BlockPatternsList` component. It also moves the search functionality for filtering pattern data out of the template-part block and into the utils directory where it can be reused.

## Testing Instructions




1. Open the site editor and edit a template with a query loop (or insert a query loop).
2. Use the "Replace" button in the query loop block toolbar
3. Confirm you see a modal with a searchable list of patterns that can be used in the query loop block.
4. Use the search and confirm the list is filtered correctly.
5. Select a pattern and confirm that it correctly replaces the content of the query post block.
6. Confirm that unit tests pass.

## Screenshots or screencast 

| Before  | After |
| ------------- | ------------- |
| <video src="https://user-images.githubusercontent.com/1464705/214386086-c1a56860-288e-49fd-bb4d-681473a5fc2a.mp4">  | <video src="https://user-images.githubusercontent.com/1464705/214386098-69bb5543-8f6c-49f6-b596-0b35d735fad1.mp4">  |
